src/main/java/org/wso2/carbon/securevault/aws/secret/repository/AWSSecretRepository.java

```diff
@@ -0,0 +1,307 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.securevault.aws.secret.repository;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.securevault.aws.common.AWSSecretManagerClient;
+import org.wso2.carbon.securevault.aws.common.AWSVaultUtils;
+import org.wso2.carbon.securevault.aws.exception.AWSVaultException;
+import org.wso2.securevault.CipherFactory;
+import org.wso2.securevault.CipherOperationMode;
+import org.wso2.securevault.DecryptionProvider;
+import org.wso2.securevault.EncodingType;
+import org.wso2.securevault.definition.CipherInformation;
+import org.wso2.securevault.keystore.IdentityKeyStoreWrapper;
+import org.wso2.securevault.keystore.KeyStoreWrapper;
+import org.wso2.securevault.keystore.TrustKeyStoreWrapper;
+import org.wso2.securevault.secret.SecretRepository;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.ALGORITHM;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.DEFAULT_ALGORITHM;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.DELIMITER;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.ENCRYPTION_ENABLED;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.KEY_STORE;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.REGEX;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.TRUSTED;
+
+/**
+ * AWS secret repository. This class is to facilitate the use of AWS Secrets Manager as an external vault
+ * for the Carbon Secure Vault.
+ */
+public class AWSSecretRepository implements SecretRepository {
+
+    private static final Log log = LogFactory.getLog(AWSSecretRepository.class);
+
+    private SecretRepository parentRepository;
+    // Secret Client used to retrieve secrets from AWS Secrets Manager Vault.
+    private SecretsManagerClient secretsClient;
+    private IdentityKeyStoreWrapper identityKeyStoreWrapper;
+    private TrustKeyStoreWrapper trustKeyStoreWrapper;
+    private DecryptionProvider baseCipher;
+    private boolean encryptionEnabled;
+
+    public AWSSecretRepository(IdentityKeyStoreWrapper identityKeyStoreWrapper,
+                               TrustKeyStoreWrapper trustKeyStoreWrapper) {
+
+        this.identityKeyStoreWrapper = identityKeyStoreWrapper;
+        this.trustKeyStoreWrapper = trustKeyStoreWrapper;
+    }
+
+    public AWSSecretRepository() {
+
+    }
```

Comment:
Do we need this empty constructor?

Reply:
Yes, this is needed because the AWSSecretRepository is initialized in the AWSSecretCallbackHandler without passing the keystore parameters into the constructor.